PayPal

lolux opened the pull request:
Ready to roll upon verification that the cookie name is stable and not
a hash specific to my account.

Commit:
PayPal

Won’t work until workaround for httpOnly cookies is in play.

Commit:
Merge branch 'site-config-PayPal' into develop

Commit:
PayPal 1

Log in with homepage form URL

Commit:
PayPal

Ready to roll upon verification that the cookie name is stable and not
a hash specific to my account.

josephwegner commented:
Almost there:

Waltz logo shows up even when you're logged in at:
- https://www.paypal.com/us/webapps/mpp/paypal-fees

josephwegner commented:
login/logout are good, though.  If you extend the `check` to match the alternative header on the fees page, this should be good to go.  Thanks!

Commit:
Extend check, add match

Extend check to account for the old-style and new-style PayPal pages.
Add Match to exclude Waltz from shopping.paypal.com and
paypal-community.com.

(Investigating why paypay-community.com displays Waltz button. It
should not since the site object match pattern should exclude it.)

Commit:
Remove paypal-community.com match

Not needed. Will open an issue to explain the situation.

lolux commented:
Thanks @josephwegner for testing. I've corrected the `check` and hidden the button on shopping.paypal.com and developer.paypal.com.

I need advice on one more usage case I discovered in testing: how to handle log ins via the [log in with PayPal](https://developer.paypal.com/docs/integration/direct/identity/log-in-with-paypal/) service. 

E.g., when you press the "log in or register" link on https://www.paypal-community.com, you are taken to a PayPal login page on which the Waltz login button appears (as it should). Waltz will log you in; however, since the `formURL` of this page (i.e., https://www.paypal.com/us/cgi-bin/webscr?SESSION=...) is different than the default PayPal login form's `formURL` (i.e., https://www.paypal.com/us/cgi-bin/webscr?cmd=_login-submit), Waltz logs you in but does not redirect you back to paypal-community.com.

I'm not sure how to handle a dynamic `formURL`. 

For the time being, we could add this to the end of the `match` pattern to exclude Waltz from appearing in this use case until we figure out a solution: `(?!/cgi-bin/webscr\\?cmd=_flow)`.

Commit:
Adjust match parameter

Exclude developer.paypal.com and sign in requests via “sign in with
PayPal” service (on the latter: temporarily until workaround is ready).
